File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.cs
using System;

class Solution
{
    bool IsPrime(int num)
    {
        for (int i = 2; i < Math.Sqrt(num) + 1; i++)
        {
            if (num % i == 0)
                return false;
        }
        return true;
    }

    public int solution(int[] nums)
    {
        int answer = 0;

        for (int i = 0; i < nums.Length - 2; i++)
        {
            for (int j = i + 1; j < nums.Length - 1; j++)
            {
                for (int k = j + 1; k < nums.Length; k++)
                {
                    answer += (IsPrime(nums[i] + nums[j] + nums[k])) ? 1 : 0;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.py
def IsPrime(num):
    for i in range(2, int(num**0.5)+1):
        if num % i == 0:
            return False
    return True

from itertools import combinations
def solution(nums):
    return sum(1 for comb in combinations(nums, 3) if IsPrime(sum(comb)))